src/providers/ipa/ipa_access.c

```diff
@@ -347,29 +344,10 @@ static void ipa_fetch_hbac_services_done(struct tevent_req *subreq)
     }
 
     /* Get the ipa_host attrs */
-    state->ipa_host = NULL;
-    ipa_hostname = dp_opt_get_cstring(state->ipa_options, IPA_HOSTNAME);
-    if (ipa_hostname == NULL) {
-        DEBUG(SSSDBG_CRIT_FAILURE,
-              "Missing ipa_hostname, this should never happen.\n");
-        ret = EINVAL;
-        goto done;
-    }
-
-    for (i = 0; i < state->hosts->entry_count; i++) {
-        ret = sysdb_attrs_get_string(state->hosts->entries[i], SYSDB_FQDN,
-                                     &hostname);
-        if (ret != EOK) {
-            DEBUG(SSSDBG_CRIT_FAILURE, "Could not locate IPA host\n");
-            goto done;
-        }
-
-        if (strcasecmp(hostname, ipa_hostname) == 0) {
-            state->ipa_host = state->hosts->entries[i];
-            break;
-        }
-    }
-
+    ret = ipa_get_host_attrs(state->ipa_options,
+                             state->hosts->entry_count,
+                             state->hosts->entries,
+                             &state->ipa_host);
```

Comment:
The return value check is strange, the function returns `ret` but the check tests the `state->ipa_host` validity. I would expect that the function signals its result via the return value.